diff --git a/src/modules/blog/domain/entities/article.entity.ts b/src/modules/blog/domain/entities/article.entity.ts
index 790f3e5..13ef987 100644
--- a/src/modules/blog/domain/entities/article.entity.ts
+++ b/src/modules/blog/domain/entities/article.entity.ts
@@ -14,7 +14,15 @@ export interface CreateArticleParams {
   image: string | null;
   contenu: string;
   datePublication: Date | null;
-  statut?: StatutArticle;
+  statut: StatutArticle | null;
+}
+
+export interface UpdateArticleParams {
+  categorieId: CategorieId;
+  nom: string;
+  image: string | null;
+  contenu: string;
+  datePublication: Date | null;
 }
 
 export class Article {
@@ -74,6 +82,14 @@ export class Article {
     return this._statut;
   }
 
+  update(params: UpdateArticleParams): void {
+    this._categorieId = params.categorieId;
+    this._nom = params.nom;
+    this._contenu = params.contenu;
+    this._image = params.image;
+    this._datePublication = params.datePublication
+  }
+
   publier(): void {
     this._statut = this._statut.transitionnerVers(StatutArticleType.PUBLIE);
     this._datePublication = this._datePublication ?? new Date();
diff --git a/src/modules/blog/domain/entities/categorie.entity.ts b/src/modules/blog/domain/entities/categorie.entity.ts
index 989235f..b37b891 100644
--- a/src/modules/blog/domain/entities/categorie.entity.ts
+++ b/src/modules/blog/domain/entities/categorie.entity.ts
@@ -5,7 +5,11 @@ export interface CreateCategorieParams {
   id: CategorieId;
   slug: Slug;
   nom: string;
-}
+};
+
+export interface UpdateCreateParams {
+  nom: string;
+};
 
 export class Categorie {
   private constructor(
@@ -29,4 +33,8 @@ export class Categorie {
   get nom(): string {
     return this._nom;
   }
+
+  update(params: UpdateCreateParams): void {
+    this._nom = params.nom;
+  }
 }
diff --git a/src/modules/blog/domain/value-objects/categorie-id.value-object.ts b/src/modules/blog/domain/value-objects/categorie-id.value-object.ts
index 1eacf44..fdde28b 100644
--- a/src/modules/blog/domain/value-objects/categorie-id.value-object.ts
+++ b/src/modules/blog/domain/value-objects/categorie-id.value-object.ts
@@ -8,6 +8,10 @@ export class CategorieId {
     return new CategorieId(value);
   }
 
+  static generate(): CategorieId {
+    return new CategorieId(crypto.randomUUID()); 
+  }
+
   toString(): string {
     return this.value;
   }
diff --git a/src/modules/blog/infrastructure/repositories/prisma-article.repository.ts b/src/modules/blog/infrastructure/repositories/prisma-article.repository.ts
index ada692f..968440b 100644
--- a/src/modules/blog/infrastructure/repositories/prisma-article.repository.ts
+++ b/src/modules/blog/infrastructure/repositories/prisma-article.repository.ts
@@ -7,6 +7,7 @@ import { ArticleId } from 'src/modules/blog/domain/value-objects/article-id.valu
 import { Slug } from 'src/shared/domain/value-objects/slug/slug.value-object.js';
 import { CategorieId } from 'src/modules/blog/domain/value-objects/categorie-id.value-object.js';
 import { handleUniqueConstraintError } from 'src/shared/infrastructure/prisma-error-handler.js';
+import { handleForeignKeyConstraintViolation } from '../errors/articles-categorie-id-fkey-error-handler.js';
 
 @Injectable()
 export class PrismaArticleRepository implements ArticleRepository {
@@ -22,6 +23,7 @@ export class PrismaArticleRepository implements ArticleRepository {
       });
     } catch (error)  {
       handleUniqueConstraintError(error, data.slug);
+      handleForeignKeyConstraintViolation(error, data.categorieId);
       throw error;
     }
     
